src/hash.cr

```diff
@@ -1,3 +1,5 @@
+# A `Hash` represents a mapping of keys to values. 
+# Check [official docs](http://crystal-lang.org/docs/syntax_and_semantics/literals/hash.html) for the basics.
```

Comment:
I'm not sure if putting a link in the introduction is a good idea or not. We don't duplicate the introduction by putting a link there, but it takes another click for users to check the documentation.